hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/hedera/simulator/BlockNodeSimulatorController.java

```diff
@@ -0,0 +1,395 @@
+     *
+     * @param index the index of the simulated block node (0-based)
+     */
+    public void resetResponses(int index) {
+        if (index >= 0 && index < simulatedBlockNodes.size()) {
+            simulatedBlockNodes.get(index).resetResponses();
+            log.info("Reset all responses on simulator {} to default behavior", index);
+        } else {
+            log.error("Invalid simulator index: {}, valid range is 0-{}", index, simulatedBlockNodes.size() - 1);
+        }
+    }
+
+    /**
+     * Get the number of simulated block nodes being controlled.
+     *
+     * @return the number of simulated block nodes
+     */
+    public int getSimulatorCount() {
+        return simulatedBlockNodes.size();
+    }
+
+    /**
+     * Shutdown all simulated block nodes to simulate connection drops.
+     * The servers can be restarted using {@link #restartAllSimulators()}.
+     */
+    public void shutdownAllSimulators() {
+        shutdownSimulatorPorts.clear();
+        for (int i = 0; i < simulatedBlockNodes.size(); i++) {
+            SimulatedBlockNodeServer server = simulatedBlockNodes.get(i);
+            int port = server.getPort();
+            shutdownSimulatorPorts.put(i, port);
+            server.stop();
+        }
+        log.info("Shutdown all {} simulators to simulate connection drops", simulatedBlockNodes.size());
+    }
+
+    /**
+     * Shutdown a specific simulated block node to simulate a connection drop.
+     * The server can be restarted using {@link #restartSimulator(int)}.
+     *
+     * @param index the index of the simulated block node (0-based)
+     */
+    public void shutdownSimulator(int index) {
+        if (index >= 0 && index < simulatedBlockNodes.size()) {
+            SimulatedBlockNodeServer server = simulatedBlockNodes.get(index);
+            int port = server.getPort();
+            shutdownSimulatorPorts.put(index, port);
+            server.stop();
+            log.info("Shutdown simulator {} on port {} to simulate connection drop", index, port);
+        } else {
+            log.error("Invalid simulator index: {}, valid range is 0-{}", index, simulatedBlockNodes.size() - 1);
+        }
+    }
+
+    /**
+     * Restart all previously shutdown simulated block nodes.
+     * This will recreate the servers on the same ports they were running on before shutdown.
+     *
+     * @throws IOException if a server fails to start
+     */
+    public void restartAllSimulators() throws IOException {
+        List<Integer> successfullyRestarted = new ArrayList<>();
+
+        for (Map.Entry<Integer, Integer> entry : shutdownSimulatorPorts.entrySet()) {
+            int index = entry.getKey();
+            int port = entry.getValue();
+
+            if (index >= 0 && index < simulatedBlockNodes.size()) {
+                // Create a new server on the same port
+                SimulatedBlockNodeServer newServer = new SimulatedBlockNodeServer(port);
+                newServer.start();
```

Comment:
Strikes me as odd that `restartAllSimulators()` doesn't call `restartSimulator()` 🤔